pkg/login/social/google_oauth.go

```diff
@@ -114,7 +114,7 @@ func (s *SocialGoogle) extractFromAPI(ctx context.Context, client *http.Client)
 }
 
 func (s *SocialGoogle) AuthCodeURL(state string, opts ...oauth2.AuthCodeOption) string {
-	if s.features.IsEnabled(featuremgmt.FlagAccessTokenExpirationCheck) {
+	if s.features.IsEnabled(featuremgmt.FlagAccessTokenExpirationCheck) && s.useRefreshToken {
```

Comment:
Do we have any plans on removing the feature toggle and only check `useRefreshToken`?

Reply:
Correct, that's the plan! (having a setting by auth provider, instead of having a global setting)